pkg/server/container_reset_windows.go

```diff
@@ -0,0 +1,177 @@
+//go:build windows
+
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package server
+
+import (
+	"encoding/json"
+	"net/url"
+	"time"
+
+	"github.com/containerd/containerd"
+	"github.com/containerd/containerd/containers"
+	"github.com/containerd/containerd/log"
+	"github.com/containerd/containerd/oci"
+	cio "github.com/containerd/cri/pkg/server/io"
+	"github.com/containerd/cri/pkg/store"
+	containerstore "github.com/containerd/cri/pkg/store/container"
+	sandboxstore "github.com/containerd/cri/pkg/store/sandbox"
+	"github.com/containerd/typeurl"
+	"github.com/pkg/errors"
+	"golang.org/x/net/context"
+	runtime "k8s.io/cri-api/pkg/apis/runtime/v1alpha2"
+)
+
+func (c *criService) resetContainer(ctx context.Context, container containerstore.Container) (retErr error) {
+	var (
+		id     = container.ID
+		meta   = container.Metadata
+		config = meta.Config
+		entity = log.G(ctx).WithField("containerID", id)
+	)
+
+	// Return error if container is neither created nor exited
+	switch state := container.Status.Get().State(); state {
+	case runtime.ContainerState_CONTAINER_CREATED:
+		entity.Debugf("container is already in created state")
+		return nil
+	case runtime.ContainerState_CONTAINER_EXITED:
+	default:
+		return errors.Errorf("container is in invalid state %q", criContainerStateToString(state))
+	}
```

Comment:
we check this later in `setContainerResetting` do we need to check again here?

Reply:
This check is primarily to short circuit the reset operation if the container hasnt been started, but `setContainerResetting` errors regardless of state. 
I can try relying on just `setContainerResetting` and see if thats cleaner if you want?

Reply:
I see. I think you should keep it as is. We should rely on an update call to the status to validate the state.